components/scream/src/physics/zm/atmosphere_deepconvection.cpp

```diff
@@ -0,0 +1,165 @@
+  Real*** fracis = &temp;
+ 
+  zm_main_f90(*m_raw_ptrs_out["lchnk"], *m_raw_ptrs_out["ncol"], m_raw_ptrs_out["t"], 
+  	      m_raw_ptrs_out["qh"], m_raw_ptrs_out["prec"], m_raw_ptrs_out["jctop"], 
+              m_raw_ptrs_out["jcbot"], m_raw_ptrs_out["pblh"], m_raw_ptrs_out["zm"], 
+              m_raw_ptrs_out["geos"], m_raw_ptrs_out["zi"], m_raw_ptrs_out["qtnd"], 
+              m_raw_ptrs_out["heat"], m_raw_ptrs_out["pap"], m_raw_ptrs_out["paph"], 
+              m_raw_ptrs_out["dpp"], *m_raw_ptrs_out["delt"], m_raw_ptrs_out["mcon"], 
+	      m_raw_ptrs_out["cme"], m_raw_ptrs_out["cape"], m_raw_ptrs_out["tpert"], 
+              m_raw_ptrs_out["dlf"], m_raw_ptrs_out["plfx"], m_raw_ptrs_out["zdu"], 
+	      m_raw_ptrs_out["rprd"], m_raw_ptrs_out["mu"], m_raw_ptrs_out["md"], 
+              m_raw_ptrs_out["du"], m_raw_ptrs_out["eu"], m_raw_ptrs_out["ed"], 
+	      m_raw_ptrs_out["dp"], m_raw_ptrs_out["dsubcld"], m_raw_ptrs_out["jt"], 
+	      m_raw_ptrs_out["maxg"], m_raw_ptrs_out["ideep"], *m_raw_ptrs_out["lengath"], 
+              m_raw_ptrs_out["ql"], m_raw_ptrs_out["rliq"], m_raw_ptrs_out["landfrac"], 
+              m_raw_ptrs_out["hu_nm1"], m_raw_ptrs_out["cnv_nm1"], m_raw_ptrs_out["tm1"], 
+              m_raw_ptrs_out["qm1"], &m_raw_ptrs_out["t_star"], &m_raw_ptrs_out["q_star"], 
+              m_raw_ptrs_out["dcape"], m_raw_ptrs_out["q"], &m_raw_ptrs_out["tend_s"], 
+              &m_raw_ptrs_out["tend_q"], &m_raw_ptrs_out["cld"], m_raw_ptrs_out["snow"], 
+              m_raw_ptrs_out["ntprprd"], m_raw_ptrs_out["ntsnprd"], 
+              &m_raw_ptrs_out["flxprec"], &m_raw_ptrs_out["flxsnow"], 
+              *m_raw_ptrs_out["ztodt"], m_raw_ptrs_out["pguall"], m_raw_ptrs_out["pgdall"], 
+              m_raw_ptrs_out["icwu"], *m_raw_ptrs_out["ncnst"], fracis); 
+  m_current_ts += dt;
+  for (int i = 0; i < zm_inputs.size(); i++){
+  	m_zm_fields_out.at(zm_inputs[i]).get_header().get_tracking().update_time_stamp(m_current_ts);
+  }
+}
+// =========================================================================================
+void ZMDeepConvection::finalize()
+{
+  zm_finalize_f90 ();
+}
+// =========================================================================================
+void ZMDeepConvection::register_fields (FieldRepository<Real, device_type>& field_repo) const {
+   for (auto& fid : m_required_fields) {
+     field_repo.register_field(fid);
+   }
+   for (auto& fid : m_computed_fields) {
+     field_repo.register_field(fid);
+   }
+ }
+
+void ZMDeepConvection::set_required_field_impl (const Field<const Real, device_type>& f) {
+  // @Meredith: Diff between add_me_as_a_customer and get_tracking().add_customer? 
+  
+  // Store a copy of the field. We need this in order to do some tracking checks
+  // at the beginning of the run call. Other than that, there would be really
+  // no need to store a scream field here; we could simply set the view ptr
+  // in the Homme's view, and be done with it.
```

Comment:
This comment seems copied from HommeDynamics. Better delete it, to not cause confusion.